### PR TITLE
build(typescript): Migrated change detection to typescript

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: tools/broccoli/broccoli-typescript.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ class DiffingTSCompiler implements DiffingBroccoliPlugin {
   private tsService: ts.LanguageService;
 
   static includeExtensions = ['.ts'];
-  static excludeExtensions = ['.d.ts'];
+  static excludeExtensions = [];
```
Comment (caitp) [inline tools/broccoli/broccoli-typescript.ts]: there's a fix for this bug waiting in #1752, or is the .d exclusion removed for another reason?

Comment (vsavkin) [inline tools/broccoli/broccoli-typescript.ts]: This is temporary. Once your fix is merged, I will rebase on top of master.

Comment (caitp) [inline tools/broccoli/broccoli-typescript.ts]: done

File: modules/angular2/src/change_detection/binding_record.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  ast: AST;
+
+  implicitReceiver: any;  // number | DirectiveIndex
+  elementIndex: number;
+  propertyName: string;
+  setter: SetterFn;
+
+  directiveRecord: DirectiveRecord;
+
+  constructor(mode: string, implicitReceiver: any, ast: AST, elementIndex: number,
```
Comment (alexeagle) [inline modules/angular2/src/change_detection/binding_record.ts]: property parameter?

File: modules/angular2/src/change_detection/change_detection.ts [hunk 3]
```diff
@@ -9,43 +9,40 @@ import {BindingRecord} from './binding_record';
...
 /**
  * Structural diffing for `Iterable` types such as `Array`s.
  *
  * @exportedAs angular2/pipes
  */
-export var iterableDiff:List<PipeFactory> = [
-  new IterableChangesFactory(),
-  new NullPipeFactory()
-];
+export var iterableDiff: List <
```
Comment (alexeagle) [inline modules/angular2/src/change_detection/change_detection.ts]: @mprobst aaaaaaahhhh

Comment (mprobst) [inline modules/angular2/src/change_detection/change_detection.ts]: This has been fixed, please make sure to upgrade your clang-format to the latest version.

**Pull request merged**